src/test/javascript/spec/component/quiz-exercise/quiz-exercise-update.component.spec.ts

```diff
@@ -635,16 +635,14 @@ describe('QuizExerciseUpdateComponent', () => {
 
             it('should set event return value to translate if not canDeactivate', () => {
                 comp.pendingChangesCache = true;
-                const ev = { returnValue: undefined };
-                comp.unloadNotification(ev);
-                expect(ev.returnValue).toBe('pendingChanges');
+                const canDeactivate = comp.canDeactivate();
+                expect(canDeactivate).toBeFalse();
             });
 
             it('should not set event return value to translate if  canDeactivate', () => {
                 comp.pendingChangesCache = false;
-                const ev = { returnValue: undefined };
-                comp.unloadNotification(ev);
-                expect(ev.returnValue).toBeUndefined();
+                const canDeactivate = comp.canDeactivate();
+                expect(canDeactivate).toBeTrue();
             });
```

Comment:
Without the HostListener, there's no event here anymore. Can you update the test descriptions? (e.g. should/should not deactivate with pending changes)